components/eamxx/src/physics/dp/impl/dp_iop_setinitial_impl.hpp

```diff
@@ -12,11 +12,203 @@ namespace dp {
+        for (Int j = 0; j < np; ++j) {
+          for (Int i = 0; i < np; ++i) {
+
+          // Find level where tobs is no longer zero
+          Int thelev=-1;
+          if (nstep <= 1) {
+            Kokkos::parallel_reduce(
+              Kokkos::TeamVectorRange(team, plev_packs), [&] (int plev_pack, Int& pmin) {
+              auto zmask = tobs(plev_pack) != 0;
+              if (zmask.any()) {
+                pmin = plev_pack;
+              }
+            }, Kokkos::Min<Int>(thelev));
+          }
+
+          if (nstep <= 1) {
+            Kokkos::parallel_for(
+              Kokkos::TeamVectorRange(team, thelev+1), [&] (int k) {
+              auto zmask = k < thelev ? Smask(true) : tobs(k) == 0;
+              vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                if (zmask[p]) {
+                  tobs(k)[p] = elem.m_forcing.m_ft(ie,i,j,k)[p];
+                  qobs(k)[p] = tracers.Q(ie,0,i,j,k)[p]; // Tracer index 0 is qobs
+                }
+              }
+            });
+          }
+          else {
+            Kokkos::parallel_for(
+              Kokkos::TeamVectorRange(team, plev_packs), [&] (int k) {
+              // Ekat packs don't know how to assign themselves to KokkosKernels::Batched::Experimental::SIMD
+              vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                tobs(k)[p] = elem.m_forcing.m_ft(ie,i,j,k)[p];
+                qobs(k)[p] = tracers.Q(ie,0,i,j,k)[p];
+              }
+            });
+          }
+
+          if (nstep == 0) {
+            if (scm_zero_non_iop_tracers) {
+              for (Int cix = 0; cix < pcnst; ++cix) {
+                Kokkos::parallel_for(
+                  Kokkos::TeamVectorRange(team, plev_packs), [&] (int k) {
+                  tracers.Q(ie, cix, i, j, k) = qmin(cix);
+                });
+              }
+            }
+
+            Kokkos::parallel_for(
+              Kokkos::TeamVectorRange(team, thelev, plev_packs), [&] (int k) {
+              auto zmask = k > thelev ? Smask(true) : tobs(k) != 0;
+              if (have_t) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  if (zmask[p]) {
+                    elem.m_forcing.m_ft(ie,i,j,k)[p] = tobs(k)[p];
+                  }
+                }
+              }
+              if (have_q) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  if (zmask[p]) {
+                    tracers.Q(ie,0,i,j,k)[p] = qobs(k)[p];
+                  }
+                }
+              }
+            });
+
+            if (have_ps) {
+              elem.m_state.m_ps_v(ie, 0, i, j) = psobs; // what is [NUM_TIME_LEVELS]?
+            }
+
+            Kokkos::parallel_for(
+              Kokkos::TeamVectorRange(team, plev_packs), [&] (int k) {
+              if (have_u) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  elem.m_state.m_v(ie, 0, 0, i, j, k)[p] = uobs(k)[p]; // [2] is 0?
```

Comment:
Wasn't sure about the [2] dimension in 
 `ExecViewManaged<Scalar * [NUM_TIME_LEVELS][2][NP][NP][NUM_LEV]>;` for `m_v`.

Reply:
that's for the 2 components of the horiz velocity.